doc/tutorial/mpi.rst

```diff
@@ -0,0 +1,115 @@
+********************
+
+Step **(12)** uses the Arbor-built-in :py:class:`MPI communicator <arbor.mpi_comm>`, which identical to the
+``COMM_WORLD`` communicator you'll know if you are familiar with MPI. The :py:class:`arbor.context` takes it as an
+argument to the ``mpi`` parameters. Note that you can also pass in communicators created with ``mpi4py``.
+We print both the communicator and context to observe how Arbor configures their defaults.
+
+.. code-block:: python
+
+   # (12) Create an MPI communicator, and use it to create a hardware context
+   comm = arbor.mpi_comm()
+   print(comm)
+   context = arbor.context(mpi=comm)
+   print(context)
+
+The execution
+*************
+
+Step **(16)** runs the simulation. Since we have more cells this time, which are connected in series, it will take some time for the action potential to propagate. In the :ref:`ring network <tutorialnetworkring>` we could see it takes about 5 ms for the signal to propgate through one cell, so let's set the runtime to ``5*ncells``.
+
+.. code-block:: python
+
+   # (16) Run simulation
+   sim.run(ncells*5)
+   print('Simulation finished')
+
+An important change in the execution is how the script is run. Whereas normally you run the Python script by passing it as an argument to the ``python`` command, you need to use ``srun`` or ``mpirun`` (depending on your MPI distribution) to execute a number of jobs in parallel. You can still execute the script using ``python``, but then MPI will not execute on more than one node.
+
+From the commandline, we can run the script using ``mpirun`` or ``srun`` and specify the number of ranks (``NRANKS``) or nodes. Arbor will spread the cells evenly over the ranks, so with ``NRANKS`` set to 5, we'd be spreading the 500 cells over 5 nodes, calculating 100 cells each.
+
+.. code-block::
+
+   srun -n NRANKS python network_ring_mpi.py
+   mpirun -n NRANKS python network_ring_mpi.py
+
+The results
+***********
+
+Before we execute the simulation, we have to understand how Arbor distributes the computational load over the ranks
+and executes the rest of the script. You can visualize this as a fork from the moment :py:func:`arbor.simulation.run`
+is reached. Here the computational load will be distributed and the subsequent execution of the script be different.
+This is relevant as to the collection of results: these are not gathered for you. Remember that in step **(14)** we
+store the handles to the probes; these referred to particular ``gid``s. The ``gid``s are now distributed, so on one
+node, the script will not find the cell referred to by the handle and therefore return an empty list (no results were found).
+
+In step **(18)** we check if the list has a nonzero length, so we can collect the results generated on this particular node. Since Arbor does not collect the results automatically, we have to write the results to disk and analyse them later. We generate a random string to use in the filename where the result is stored.
```

Comment:
Why do we need a random string? It's more useful to use the rank id, no?

Reply:
Old habits!